pages/visualizations/overview/issues_over_time.py

```diff
@@ -127,96 +127,125 @@ def issues_over_time_graph(repolist, timer_pings, interval):
+    closed_range = pd.to_datetime(df["closed"]).dt.to_period(interval).value_counts().sort_index()
+    df_closed = closed_range.to_frame().reset_index().rename(columns={"index": "Date"})
+    df_closed["Date"] = pd.to_datetime(df_closed["Date"].astype(str).str[:period_slice])
+
+    if interval == "M":
+        df_created["Date"] = df_created["Date"].dt.strftime("%Y-%m-01")
+        df_closed["Date"] = df_closed["Date"].dt.strftime("%Y-%m-01")
+    elif interval == "Y":
+        df_created["Date"] = df_created["Date"].dt.strftime("%Y-01-01")
+        df_closed["Date"] = df_closed["Date"].dt.strftime("%Y-01-01")
+
+    # first and last elements of the dataframe are the
+    # earliest and latest events respectively
+    earliest, latest = df.iloc[0]["created"], df.iloc[-1]["created"]
+
+    # beginning to the end of time by the specified interval
+    dates = pd.date_range(start=earliest, end=latest, freq="D", inclusive="both")
+
+    # df for open issues for time interval
+    df_open = dates.to_frame(index=False, name="Date")
 
-    # df for line chart
-    df_open = make_open_df(df_issues)
-    if df_open is None:
-        logging.debug("ISSUES_OVER_TIME_VIZ - NO DATA AVAILABLE")
-        return nodata_graph, dash.no_update
+    # aplies function to get the amount of open issues for each day
+    df_open["Open"] = df_open.apply(lambda row: get_open(df, row.Date), axis=1)
 
-    # reset index to be ready for plotly
-    df_issues = df_issues.reset_index()
+    df_open["Date"] = df_open["Date"].dt.strftime("%Y-%m-%d")
 
     # time values for graph
     x_r, x_name, hover, period = get_graph_time_values(interval)
 
     # graph generation
-    if df_issues is not None:
-        fig = go.Figure()
-        fig.add_histogram(
-            x=df_issues["closed"].dropna(),
-            histfunc="count",
-            name="closed",
-            opacity=0.75,
-            hovertemplate=hover + "<br>Closed: %{y}<br>" + "<extra></extra>",
+    fig = go.Figure()
+    fig.add_bar(
+        x=df_created["Date"],
+        y=df_created["created"],
+        opacity=0.75,
+        hovertemplate=hover + "<br>Created: %{y}<br>" + "<extra></extra>",
+        offsetgroup=0,
+        name="Issues Created",
+    )
+    fig.add_bar(
+        x=df_closed["Date"],
+        y=df_closed["closed"],
+        opacity=0.6,
+        hovertemplate=hover + "<br>Closed: %{y}<br>" + "<extra></extra>",
+        offsetgroup=1,
+        name="Issues Closed",
+    )
+    fig.update_xaxes(
+        showgrid=True,
+        ticklabelmode="period",
+        dtick=period,
+        rangeslider_yaxis_rangemode="match",
+        range=x_r,
+    )
+    fig.update_layout(
+        xaxis_title=x_name,
+        yaxis_title="Number of Issues",
+        bargroupgap=0.1,
+        margin_b=40,
+    )
+    fig.add_trace(
+        go.Scatter(
+            x=df_open["Date"],
+            y=df_open["Open"],
+            mode="lines",
+            name="Issues Actively Open",
```

Comment:
can drop the 'actively' since we're not measuring the activity of the open issue

Reply:
Maybe we can choose a different adjective. I want to distinguish between an issue being "opened" during that time vs being actively (cant think of a different word atm) open. Ie if we are are looking sometime in Oct 2022 many of the issues open could be from months before